### PR TITLE
pass the context into Close calls

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

File: sql/analyzer/releaser.go [hunk 1]
```diff
@@ -75,16 +75,16 @@ type releaseIter struct {
 func (i *releaseIter) Next() (sql.Row, error) {
 	row, err := i.child.Next()
 	if err != nil {
-		_ = i.Close()
+		_ = i.Close(nil)
```
Comment (reltuk) [inline sql/analyzer/releaser.go:78]: `Next(*sql.Context) (sql.Row, error)` wen?!

File: sql/row_test.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ func TestRowsToRowIterEmpty(t *testing.T) {
 	require.Equal(io.EOF, err)
 	require.Nil(r)
 
-	err = iter.Close()
+	err = iter.Close(nil)
```
Comment (reltuk) [inline sql/row_test.go:36]: Make a fake Context.

**Review by reltuk — APPROVED** [review]
LGTM! Some questions.

**Review by zachmu — APPROVED** [review]
LG

File: sql/analyzer/releaser.go (hunk 1 above)
Comment (bheni) [reply sql/analyzer/releaser.go]: I thought about it.  It's only going to get worse.  Should probably do it soon.

**Pull request merged**